CalendarAppTests/CalendarEventHandlerTests.swift

```diff
@@ -0,0 +1,329 @@
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent, longEvent])
+        
+        handler.deleteEvent(longEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(yesterday), [earlyEvent])
+    }
+    
+    func testAddEventAfterDelete() {
+        let event = Event()
+        handler.addEventsFromArray([currentEvent, event], Date())
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent, event])
+        handler.deleteEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event])
+
+        handler.addNewEvent(currentEvent)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [event, currentEvent])
+    }
+    
+    func testUpdateEvent() {
+        handler.addEventsFromArray([currentEvent], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.eventTitle = "hi"
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateNonexistentEvent() {
+        handler.addEventsFromArray([], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        XCTAssertEqual(handler.getEventsForDate(Date()), [currentEvent])
+    }
+    
+    func testUpdateEventToNewDate() {
+        handler.addEventsFromArray([currentEvent], Date())
+        handler.addEventsFromArray([], tomorrow)
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.startDate = currentEvent.startDate.advanced(by: TimeInterval(86400))
+        currentEvent.endDate = currentEvent.endDate.advanced(by: TimeInterval(86400))
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [currentEvent])
+    }
+    
+    func testUpdateEventToNonexistingDate() {
+        handler.addEventsFromArray([currentEvent], Date())
+        let originalStart = currentEvent.startDate
+        let originalEnd = currentEvent.endDate
+        
+        currentEvent.startDate = currentEvent.startDate.advanced(by: TimeInterval(86400))
+        currentEvent.endDate = currentEvent.endDate.advanced(by: TimeInterval(86400))
+        handler.updateEvent(currentEvent, originalStart, originalEnd)
+        
+        XCTAssertEqual(handler.getEventsForDate(Date()), [])
+        XCTAssertEqual(handler.getEventsForDate(tomorrow), [])
+    }
+    
+    func testUpdateEventToMultipleDayEvent() {
+        handler.addEventsFromArray([currentEvent], Date())
+        handler.addEventsFromArray([], tomorrow)
```

Comment:
Yeah, having to make sure a tomorrow 'bucket' exists this way feels like a hack. If the hack can't be moved into the model or the root of it can't be easily fixed then this should have a comment explaining why this is happening imo.